pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/SierraTransformer.scala

```diff
@@ -0,0 +1,33 @@
+package uk.ac.wellcome.platform.transformer.sierra
+
+import grizzled.slf4j.Logging
+
+import uk.ac.wellcome.models.transformable.sierra.SierraBibNumber
+import uk.ac.wellcome.platform.transformer.sierra.source.SierraBibData
+
+trait Info {
+  def info: String
```

Comment:
I wonder if info could return a type of Log, that then has a toString, which we could JSONify later?

e.g.
```scala
sealed trait Log { def toString: String }
trait Info {
  def info: Log
}

case class SierraTransformerLog(bibId: String, tranformerName: string, info: string, output: String) extends Log {
  def toString = "DATA TRANSFORMED: " +
        s"BibID=${bibId}, " +
        s"Transformer=${transformerName}, " +
        s"Info=${transformInfo.info}, " +
        s"Output=${output}"
}
```

I've missed pieces out, but hopefully that gives the gist.... just a thought as we've been talking about GELP etc.

Reply:
@jamesgorrie have done something similar to what you suggested.